sklearn/decomposition/tests/test_factor_analysis.py

```diff
@@ -83,3 +86,41 @@ def test_factor_analysis():
         precision = fa.get_precision()
         assert_array_almost_equal(np.dot(cov, precision),
                                   np.eye(X.shape[1]), 12)
+
+
+@pytest.mark.parametrize("data_type, expected_type", [
+    (np.float32, np.float32),
+    (np.float64, np.float64),
+    (np.int32, np.float64),
+    (np.int64, np.float64)
+])
+def test_lda_dtype_match(data_type, expected_type):
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 20, 5, 3
+    W = rng.randn(n_components, n_features)
+    h = rng.randn(n_samples, n_components)
+    noise = rng.gamma(1, size=n_features) * rng.randn(n_samples, n_features)
+    X = np.dot(h, W) + noise
+    for method in ['randomized', 'lapack']:
+        clf = FactorAnalysis(n_components=n_components, svd_method=method)
+        clf.fit(X.astype(data_type))
+        assert clf.components_.dtype == expected_type
+
+
+
+def test_lda_numeric_consistency_float32_float64():
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 20, 5, 3
+    W = rng.randn(n_components, n_features)
+    h = rng.randn(n_samples, n_components)
+    noise = rng.gamma(1, size=n_features) * rng.randn(n_samples, n_features)
+    X = np.dot(h, W) + noise
+    for method in ['randomized', 'lapack']:
+        clf_32 = FactorAnalysis(n_components=n_components, svd_method=method, random_state=42)
+        clf_32.fit(X.astype(np.float32))
+        clf_64 = FactorAnalysis(n_components=n_components, svd_method=method, random_state=42)
+        clf_64.fit(X.astype(np.float64))
+
+        # Check value consistency between types
+        rtol = 1e-5
```

Comment:
The test breaks for rtol=1e-6, even though atol~1e-8 in the test. Should I change the numerical consistency criteria to atol=1e-6 ?